### PR TITLE
Minor fixes

### PR DESCRIPTION
**Pull request by ninedraft**
Minor style and performance fixes

### CODE REVIEW

**Review by cristaloleg — APPROVED** [review]

**Comment by amit-davidson** [comment]
Wow, thank you very much for your effort.
I'll look at all the PRs in the evening.

**Comment by cristaloleg** [comment]
Looks like this PR contains few other commits from other PRs.

**Comment by ninedraft** [comment]
> Looks like this PR contains few other commits from other PRs.

Shame on me

Fixed

File: domain/Lockset.go [hunk 1]
```diff
@@ -2,40 +2,33 @@ package domain
...
 }
 
 func (ls *Lockset) UpdateLockSet(newLocks, newUnlocks locksLasUse) {
-	if newLocks != nil {
-		for lockName, lock := range newLocks {
-			ls.ExistingLocks[lockName] = lock
-		}
-	}
 	for unlockName, _ := range newUnlocks {
-		if _, ok := ls.ExistingLocks[unlockName]; ok {
```
Comment (amit-davidson) [inline domain/Lockset.go:29]: Unlocks are never removed in your change so I would keep the first block (but still trimming that nil check). 
The algorithm works by remembering each lock's state (locked/unlocked/or nothing, of course). It means that if a mutex was unlocked at some point but later was locked again, then its latest status is locked, and the unlock status is removed. This is the formula. I hope it's more clear:

image

File: domain/Lockset.go [hunk 2]
```diff
@@ -2,40 +2,33 @@ package domain
...
 
-	if newUnlocks != nil {
-		for unlockName, unlock := range newUnlocks {
-			ls.ExistingUnlocks[unlockName] = unlock
-		}
+	for unlockName, unlock := range newUnlocks {
+		ls.ExistingUnlocks[unlockName] = unlock
 	}
+
 	for lockName, _ := range newLocks {
```
Comment (amit-davidson) [inline domain/Lockset.go:32]: Also, if possible, can you change this block as well and trim all the redundant code from it as you did to the other blocks?

**Review by amit-davidson — CHANGES_REQUESTED** [review]
Hey, first of all, I wanted to thank you for all your effort. This PR and the others.
Second, I really liked your changes so well done!
Just mind my 2 small comments and it's all good.

**Comment by amit-davidson** [comment]
Also, you can run tests locally. They are located under e2e-tests/summary_test.go.

File: domain/Lockset.go (hunk 1 above)
Comment (cristaloleg) [reply domain/Lockset.go:29]: @ninedraft idea was about 'no need to check if element is in map before deletion'. And that make sense, 'cause everything from `newUnlocks` will be removed anyway, isn't it?

Comment (amit-davidson) [reply domain/Lockset.go:29]: @cristaloleg what do you mean by "cause everything from `newUnlocks` will be removed anyway"?

Comment (cristaloleg) [reply domain/Lockset.go:29]: oh yeah, got it, sorry :)

File: domain/Lockset.go (hunk 2 above)
Comment (ninedraft) [reply domain/Lockset.go]: ✅

File: domain/Lockset.go (hunk 1 above)
Comment (ninedraft) [reply domain/Lockset.go:29]: ✅

**Comment by ninedraft** [comment]
Fixed remarks

**Comment by amit-davidson** [comment]
@ninedraft You can click the re-review button if it's ready

**Review by amit-davidson — APPROVED** [review]
Nice! Thank you for your contribution.

**Pull request merged**